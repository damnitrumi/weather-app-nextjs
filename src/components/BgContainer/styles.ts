import styled, { css } from "styled-components";
import { BgContainerProps } from ".";

export const Wrapper = styled.article<Pick<BgContainerProps, "bgSelection">>`
  ${({ bgSelection }) => css`
    display: flex;
    flex-direction: column;
    background-image: url(${bgSelection});
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    width: 100%;
    height: 100%;
    margin: 0 auto;

    @media(min-width: 640px){
      flex-direction: row;
    }

    @media(min-width: 768px){
      aspect-ratio: 16/9;
      height: auto;
      max-width: 1000px;
      border-radius: 10px;
      overflow: hidden;

    }
  `}
`;
