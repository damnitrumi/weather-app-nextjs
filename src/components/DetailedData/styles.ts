import styled, { css } from "styled-components";

export const Wrapper = styled.section`
  ${() => css`
    padding:20px;
    min-width: 200px;
    backdrop-filter: blur(10px) saturate(180%);
    -webkit-backdrop-filter: blur(10px) saturate(180%);

    @media (min-width: 640px){
      flex-grow: 2;
    }

    @media (min-width: 1024px){
      padding: 40px;
    }
  `}
`;
