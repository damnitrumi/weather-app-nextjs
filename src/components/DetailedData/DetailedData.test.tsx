import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { DetailedData, DetailedDataProps } from ".";

const props: DetailedDataProps = {
  maxTemp: 1,
  minTemp: 2,
  cloudiness: 3,
  humidity: 4,
  windSpeed: 5,
};

describe("<DetailedData />", () => {
  it("should render DetailedData and its children", () => {
    renderTheme(<DetailedData {...props} />);

    expect(
      screen.getByRole("heading", { name: "Weather Details:" }),
    ).toBeInTheDocument();

    expect(screen.getByText("Max Temperature")).toBeInTheDocument();
    expect(screen.getByText("Min Temperature")).toBeInTheDocument();
    expect(screen.getByText("Cloudiness")).toBeInTheDocument();
    expect(screen.getByText("Humidity")).toBeInTheDocument();
    expect(screen.getByText("Wind")).toBeInTheDocument();

    expect(screen.getByText("1°")).toBeInTheDocument();
    expect(screen.getByText("2°")).toBeInTheDocument();
    expect(screen.getByText("3%")).toBeInTheDocument();
    expect(screen.getByText("4%")).toBeInTheDocument();
    expect(screen.getByText("5km/h")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<DetailedData {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
