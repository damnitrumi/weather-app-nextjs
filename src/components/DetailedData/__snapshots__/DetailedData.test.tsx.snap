// Vitest Snapshot v1

exports[`<DetailedData /> > should match snapshot 1`] = `
.c0 {
  padding: 20px;
  min-width: 200px;
  -webkit-backdrop-filter: blur(10px) saturate(180%);
  backdrop-filter: blur(10px) saturate(180%);
  -webkit-backdrop-filter: blur(10px) saturate(180%);
}

@media (min-width:1024px) {
  .c0 {
    padding: 40px;
  }
}

<section
  class="c0"
>
  <h1
    class="text-white font-bold mb-8 text-[1.8rem] lg:text-[2.4rem]"
  >
    Weather Details:
  </h1>
  <ul
    class="list-none flex flex-col gap-1"
  >
    <li
      class="flex justify-between text-gray-300"
    >
      <span
        class="text-[1.4rem] lg:text-[1.6rem]"
      >
        Max Temperature
      </span>
      <span
        class="text-white text-[1.4rem] lg:text-[1.6rem]"
      >
        1
        °
      </span>
    </li>
    <li
      class="flex justify-between text-gray-300"
    >
      <span
        class="text-[1.4rem] lg:text-[1.6rem]"
      >
        Min Temperature
      </span>
      <span
        class="text-white text-[1.4rem] lg:text-[1.6rem]"
      >
        2
        °
      </span>
    </li>
    <li
      class="flex justify-between text-gray-300"
    >
      <span
        class="text-[1.4rem] lg:text-[1.6rem]"
      >
        Cloudiness
      </span>
      <span
        class="text-white text-[1.4rem] lg:text-[1.6rem]"
      >
        3
        %
      </span>
    </li>
    <li
      class="flex justify-between text-gray-300"
    >
      <span
        class="text-[1.4rem] lg:text-[1.6rem]"
      >
        Humidity
      </span>
      <span
        class="text-white text-[1.4rem] lg:text-[1.6rem]"
      >
        4
        %
      </span>
    </li>
    <li
      class="flex justify-between text-gray-300"
    >
      <span
        class="text-[1.4rem] lg:text-[1.6rem]"
      >
        Wind
      </span>
      <span
        class="text-white text-[1.4rem] lg:text-[1.6rem]"
      >
        5
        km/h
      </span>
    </li>
  </ul>
</section>
`;
